packer/setup_root.sh

```diff
@@ -84,3 +85,30 @@ helm repo add iterative https://helm.iterative.ai
 # Copy the support bundle script
 cp /home/ubuntu/.studio_install/create-support-bundle /usr/local/bin/create-support-bundle
 chmod +x /usr/local/bin/create-support-bundle
+
+# Cache Images in K3s
+
+## Air-Gap Install https://docs.k3s.io/installation/airgap#prepare-the-images-directory-and-k3s-binary
+mkdir -p /var/lib/rancher/k3s/agent/images/
+curl "https://github.com/k3s-io/k3s/releases/download/$K3S_VERSION/k3s-airgap-images-amd64.tar" -L -o /var/lib/rancher/k3s/agent/images/k3s-airgap-images-amd64.tar
+
+## Download Docker and fetch ingress-nginx related images
+curl -fsSL https://get.docker.com -o get-docker.sh
+sh get-docker.sh
+
+docker pull -q ghcr.io/iterative/studio-selfhosted/kh-klipper-cache:${kh_klipper_tag}
+docker save ghcr.io/iterative/studio-selfhosted/kh-klipper-cache:${kh_klipper_tag} -o kh-klipper.tar
+
+docker pull -q registry.k8s.io/ingress-nginx/controller:v1.7.0
+docker save registry.k8s.io/ingress-nginx/controller:v1.7.0 -o ingress-nginx-controller.tar
+
+docker pull -q registry.k8s.io/ingress-nginx/kube-webhook-certgen:v20230312-helm-chart-4.5.2-28-g66a760794
+docker save registry.k8s.io/ingress-nginx/kube-webhook-certgen:v20230312-helm-chart-4.5.2-28-g66a760794 -o ingress-nginx-controller-kube-webhook-certgen.tar
+
+mv *.tar /var/lib/rancher/k3s/agent/images/
+
+## Cleanup docker
+apt-get purge  docker* -y
+rm -rf get-docker.sh /var/lib/docker/
```

Comment:
I was playing with [skopeo](https://github.com/containers/skopeo) and [Buildah](https://github.com/containers/buildah), but I cannot find a way to export to the correct image format

Reply:
Yeah let's try to follow up on it later